February release highlights for JavaScript & TypeScript (#2333)
Azure/azure-sdk: releases/2021-02/js.md

repository: azure/azure-sdk-for-js
---

The Azure SDK team is pleased to make available the February 2021 client library release.

#### GA

- Azure Search Documents
- Azure Event Hubs

#### Updates

- Azure Storage Blob
- Azure Storage Queue
- Azure File Datalake
- Azure File Share
- Azure Identity

#### Beta

- Azure Text Analytics
- Azure Metrics Advisor
- Azure Form Recognizer
- Azure Identity
- Azure Key Vault Keys
- Azure Key Vault Admin
- Azure Storage Blob
- Azure File Datalake

> - Azure Event Grid

Reviewer: ```suggestion
- Azure Event Grid
- Azure Data Tables
```

## Installation Instructions

To install the packages, copy and paste the below into a terminal.

```bash
$> npm install @azure/search-documents
$> npm install @azure/event-hubs
$> npm install @azure/storage-blob
$> npm install @azure/storage-file-datalake
$> npm install @azure/storage-file-share
$> npm install @azure/storage-queue
$> npm install @azure/ai-text-analytics@next
$> npm install @azure/ai-metrics-advisor@next
$> npm install @azure/ai-form-recognizer@next
$> npm install @azure/identity@next
$> npm install @azure/keyvault-keys@next
$> npm install @azure/keyvault-admin@next
$> npm install @azure/storage-blob@next
$> npm install @azure/storage-file-datalake@next

> $> npm install @azure/eventgrid@next

Reviewer: ```suggestion
$> npm install @azure/eventgrid@next
$> npm install @azure/data-tables@next
```

#### GA

- Azure Search Documents
- Azure Event Hubs

#### Updates

> - Azure Storage Blob

Reviewer 2: Should these be package names?

Author: We've been using the product name each time we do these highlights 🤔 we could change it!

Reviewer 2: I don't have any strong preference here but we should see if this list adds enough value over the other parts below.

#### GA

- Azure Search Documents

#### Updates

- Azure Storage Blob
- Azure Storage Queue
- Azure File Datalake

> - Azure File Share

Reviewer 3: ```suggestion
- Azure File Share
- Azure Event Hubs
```

Author: I moved event hubs to the GA section 🌞 thanks for the update!

Author: I have pushed an update here: https://github.com/Azure/azure-sdk/pull/2333/commits/ce3ac4b7e18b5327628227ac86cd9b97405f2b55

#### Beta

> - Azure AI Text Analytics
> - Azure AI Metrics Advisor
> - Azure AI Form Recognizer

Reviewer 4: ```suggestion
- Azure Text Analytics
- Azure Metrics Advisor
- Azure Form Recognizer
```

IIRC, we use `ai-form-recognizer` as the package name, but the product name is just "Form Recognizer". Same for the others. (CC @deyaaeldeen @jeremymeng)

Author: Thank you!

Author: I have pushed an update here: https://github.com/Azure/azure-sdk/pull/2333/commits/ce3ac4b7e18b5327628227ac86cd9b97405f2b55

- Azure Identity
- Azure Key Vault Keys
- Azure Key Vault Secrets
- Azure Key Vault Certificates
- Azure Key Vault Admin

> - Azure Event Hubs

Reviewer 3: Event Hubs is a GA release, not beta.
```suggestion
```

Author: Thank you! I have pushed an update here: https://github.com/Azure/azure-sdk/pull/2333/commits/ce3ac4b7e18b5327628227ac86cd9b97405f2b55

- Azure Synapse

> - Azure Storage Blob

Reviewer 5: ```suggestion
- Azure Storage Blob
- Azure File Datalake
```

Author: Thank you! I have pushed an update here: https://github.com/Azure/azure-sdk/pull/2333/commits/ce3ac4b7e18b5327628227ac86cd9b97405f2b55

- Azure Event Grid

## Installation Instructions

To install the packages, copy and paste the below into a terminal.

```bash
$> npm install @azure/search-documents
$> npm install @azure/storage-blob
$> npm install @azure/storage-file-datalake
$> npm install @azure/storage-file-share
$> npm install @azure/storage-queue
$> npm install @azure/ai-text-analytics@next
$> npm install @azure/ai-metrics-advisor@next
$> npm install @azure/ai-form-recognizer@next
$> npm install @azure/identity@next
$> npm install @azure/keyvault-keys@next
$> npm install @azure/keyvault-secrets@next
$> npm install @azure/keyvault-certificates@next
$> npm install @azure/keyvault-admin@next

> $> npm install @azure/event-hubs@next

Reviewer 3: Event Hubs is a GA release, so we shouldn't need the `@next` tag.

Author: Thank you! I have pushed an update here: https://github.com/Azure/azure-sdk/pull/2333/commits/ce3ac4b7e18b5327628227ac86cd9b97405f2b55

$> npm install @azure/synapse@next

> $> npm install @azure/storage-blob@next

Reviewer 5: ```suggestion
$> npm install @azure/storage-blob@next
$> npm install @azure/storage-file-datalake@next
```

Author: Thank you! I have pushed an update here: https://github.com/Azure/azure-sdk/pull/2333/commits/ce3ac4b7e18b5327628227ac86cd9b97405f2b55

## Release highlights

> 

Reviewer 6: ```suggestion
### AI Metrics Advisor

- We are releasing a new  preview version for metrics advisor with AAD support feature

#### @azure/ai-metrics-advisor [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/metricsadvisor/ai-metrics-advisor/CHANGELOG.md)

###  New Features in @azure/ai-metrics-advisor@1.0.0@beta.3

- Metrics Advisor is releasing support for Azure Active Directory Authentication and support to rotate the API and Subscription Keys.

```

Author: Added! Thank you!

## Release highlights

### Event Hubs

We're releasing a new GA for the Azure Event Hubs client with new features and a major fix.

> #### @azure/event-hubs [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/CHANGELOG.md)

Reviewer 2: It is best to have change log entries use the release tags instead of master.

Author: Done! Except for one identity entry that is going out today

- `CryptographyClient` can now support "local only" mode by instantiating it with only a `JsonWebKey`, a new constructor overload was added to support this.
- Users can now get the `keyId` property from a `CryptographyClient`.

### Azure Text Analytics

We're releasing a new beta for the Text Analytics service including new features and some renamed properties and methods.

#### @azure/ai-text-analytics [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/textanalytics/ai-text-analytics/CHANGELOG.md#510-beta4-2021-02-10)

##### New Features in @azure/ai-text-analytics@5.1.0-beta.4

- A new option to control how the offset is calculated by the service. For more information, see [the Text Analytics documentation](https://docs.microsoft.com/azure/cognitive-services/text-analytics/concepts/text-offsets#offsets-in-api-version-31-preview).

##### Breaking Changes on @azure/ai-text-analytics@5.1.0-beta.4

- The healthcare entities returned by `beginAnalyzeHealthcareEntities` are now organized as a directed graph where the edges represent a certain type of healthcare relationship between the source and target entities. Edges are stored in the `relatedEntities` property.

> ### AI Metrics Advisor

Reviewer 7: ```suggestion
### Metrics Advisor
```
Do we need to have `Azure` in the front? Looks like Event Hubs heading doesn't have it.

Author: Azure products generally have Azure in front, so I will add it! Thank you for noticing this.

Author: I have updated the headers to include Azure, and I have included your change here 🌞

#### @azure/ai-text-analytics [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/textanalytics/ai-text-analytics/CHANGELOG.md#510-beta4-2021-02-10)

##### New Features in @azure/ai-text-analytics@5.1.0-beta.4

- A new option to control how the offset is calculated by the service. For more information, see [the Text Analytics documentation](https://docs.microsoft.com/azure/cognitive-services/text-analytics/concepts/text-offsets#offsets-in-api-version-31-preview).

##### Breaking Changes on @azure/ai-text-analytics@5.1.0-beta.4

- The healthcare entities returned by `beginAnalyzeHealthcareEntities` are now organized as a directed graph where the edges represent a certain type of healthcare relationship between the source and target entities. Edges are stored in the `relatedEntities` property.

### Azure Metrics Advisor

- We are releasing a new  preview version for metrics advisor with AAD support feature

#### @azure/ai-metrics-advisor [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/metricsadvisor/ai-metrics-advisor/CHANGELOG.md#100-beta3-2021-02-09)

##### New Features in @azure/ai-metrics-advisor@1.0.0@beta.3

> - Metrics Advisor is releasing support for Azure Active Directory Authentication and support to rotate the API and Subscription Keys.

Reviewer 6: I think I added more to this - 

```suggestion
- Metrics Advisor is releasing support for Azure Active Directory Authentication and support to rotate the API and Subscription Keys.

##### Breaking Changes in @azure/ai-metrics-advisor@1.0.0@beta.3
- Renamed a method `listDimensionValuesForDetectionConfig()` to `listAnomalyDimensionValues()`
- Changed the response types for update methods

```

Author: My bad! Thank you!

## Release highlights

### Azure Event Hubs

We're releasing a new GA for the Azure Event Hubs client with new features and a major fix.

#### @azure/event-hubs [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/CHANGELOG.md#540-2021-02-09)

##### New Features in @azure/event-hubs@5.4.0

- You can now specify custom endpoint address as a client constructor option to use when communicating with the Event Hubs service, which is useful when your network does not allow communicating to the standard Event Hubs endpoint.
- Added a helper method `parseEventHubConnectionString` that can be used to validate and parse a given connection string for Azure Event Hubs.

##### Major Fixes in @azure/event-hubs@5.4.0

> - Fixes an issue where the `RetryMode` enum for use when setting the RetryOptions.mode field in `EventHubConsumerClientOptions` or `EventHubClientOptions` wasn't exported by the package.

Reviewer 8: I would skip this. User does not have to change anything unless they want to change the retry mode for which there were workarounds too. See [What is a developer impacting change?](https://github.com/Azure/azure-sdk/blob/08f279b39a4c19822b1509e6b10fc3ce256607b5/docs/policies/releasenotes.md#whats-a-developer-impacting-change)

cc @chradek

Author: Removing it!

Author: Removed it here: https://github.com/Azure/azure-sdk/pull/2333/commits/7c801836c2799febab0274578673c4527cf86e6f

### Azure Identity

Identity is releasing a patch with a dependency fix and a bug fix, and a beta that includes MSAL 2.0 with PKCE support.

#### @azure/identity [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/identity/identity/CHANGELOG.md)

##### Major Fixes in @azure/identity@1.2.3 ([Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/identity/identity/CHANGELOG.md#123-2021-02-09))

> - Fixed Azure Stack support for the NodeJS version of the `InteractiveBrowserCredential`.

Reviewer 8: Can we talk about what the problem before was? If I am an Azure Stack user, I wouldnt know whether this line would apply to me or not

Author: Thoughts on this?

> Fixed a bug that appeared while using `InteractiveBrowserCredential` to authenticate against Azure Stack from NodeJS (for local apps that would ask for authentication using a browser window). Azure Stack has specific authentication requirements that were missed on our previous `InteractiveBrowserCredential` release.

> - The 'keytar' dependency has been updated to the latest version.

Reviewer 8: A dependency update is not a release highlight unless it solved a major problem for the customer. I recall the user did have some trouble which required us to do this update. Mentioning that here would help

Author: This is the original issue: https://github.com/Azure/azure-sdk-for-js/issues/13531

That issue is still open and the real solution is not trivial. This update helps but it's very minor. I'll remove the highlight entry.

##### Changes in @azure/identity@1.2.4-beta.1

> - In this beta we've updated `InteractiveBrowserCredential` to use the Auth Code Flow with PKCE rather than Implicit Grant Flow by default in the browser, to better support browsers with enhanced security restrictions.

Reviewer 8: Can we provide links for `Auth Code Flow with PKCE` and the `Implicit Grant flow`?

Author: Done here: https://github.com/Azure/azure-sdk/pull/2333/commits/7c801836c2799febab0274578673c4527cf86e6f

### Azure Key Vault

We're releasing some beta updates for the clients of the Key Vault service with small updates and breaking changes only from the last beta updates released.

#### @azure/keyvault-admin [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/keyvault/keyvault-admin/CHANGELOG.md#400-beta2-2021-02-09)

##### Changes on @azure/keyvault-admin@4.0.0-beta.2

> - Backup and restore operations will now return a `BackupResult` and `RestoreResult` as appropriate.

Reviewer 8: What was it returning before? Does the user have to change their code? If this is not a breaking change or highly impacting, perhaps we can skip mentioning it?

The points under `Major Fixes on @azure/keyvault-admin@4.0.0-beta.2` do not seem to be developer impacting changes either. They do not need to react to these changes. I would recommend removing both the `Major Fixes` and this current section here and have say just this line

> The beta releases has bug fixes to improve quality and support for custom role definitions.

See [What is a developer impacting change?](https://github.com/Azure/azure-sdk/blob/08f279b39a4c19822b1509e6b10fc3ce256607b5/docs/policies/releasenotes.md#whats-a-developer-impacting-change)

cc @maorleger

Author: It's a breaking change, but from one beta to another, since this client hasn't been GAd.

BackupResult and RestoreResult from backup/restore long running operations now contain additional information about the operation, we can say that 🤔

- Users can now get the `keyId` property from a `CryptographyClient`.

### Azure Text Analytics

We're releasing a new beta for the Text Analytics service including new features and some renamed properties and methods.

#### @azure/ai-text-analytics [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/textanalytics/ai-text-analytics/CHANGELOG.md#510-beta4-2021-02-10)

##### New Features in @azure/ai-text-analytics@5.1.0-beta.4

- A new option to control how the offset is calculated by the service. For more information, see [the Text Analytics documentation](https://docs.microsoft.com/azure/cognitive-services/text-analytics/concepts/text-offsets#offsets-in-api-version-31-preview).

##### Breaking Changes on @azure/ai-text-analytics@5.1.0-beta.4

- The healthcare entities returned by `beginAnalyzeHealthcareEntities` are now organized as a directed graph where the edges represent a certain type of healthcare relationship between the source and target entities. Edges are stored in the `relatedEntities` property.

### Azure Metrics Advisor

> - We are releasing a new  preview version for metrics advisor with AAD support feature

Reviewer 8: This should not be a bullet point

```suggestion
We are releasing a new  preview version for metrics advisor with AAD support feature
```

#### @azure/ai-metrics-advisor [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/metricsadvisor/ai-metrics-advisor/CHANGELOG.md#100-beta3-2021-02-09)

##### New Features in @azure/ai-metrics-advisor@1.0.0@beta.3

> - Metrics Advisor is releasing support for Azure Active Directory Authentication and support to rotate the API and Subscription Keys.

Reviewer 8: ```suggestion
- Metrics Advisor is releasing support for Azure Active Directory Authentication and the rotation of the API and Subscription Keys.
```

##### Breaking Changes in @azure/ai-metrics-advisor@1.0.0@beta.3

- Renamed a method `listDimensionValuesForDetectionConfig()` to `listAnomalyDimensionValues()`.
- Changed the response types for update methods.
- Added mapping of metric id to metric name in Datafeed object.

### Azure Form Recognizer

We're releasing a new beta client for the Azure Form Recognizer service with an upgrade for one TypeScript type.

#### @azure/ai-form-recognizer [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/formrecognizer/ai-form-recognizer/CHANGELOG.md#310-beta2-2021-02-09)

##### Breaking Changes in @azure/ai-form-recognizer@3.1.0-beta.2

- Reworked the type of the `appearance` field and its children of the `FormLine` type to have more specific names.

### Azure Storage

We released hot fix versions and also beta versions for new service features in Azure Storage Service API version 2020-06-10.

#### @azure/storage-blob [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-blob/CHANGELOG.md)

> ##### Major Fixes in @azure/storage-blob@12.4.1 ([Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-blob/CHANGELOG.md#1241-2021-02-03))

Reviewer 8: Do we need the changelog link at this level? Wouldnt the changelog link a few lines above suffice?

Author: We generally point to the exact version in the changelog above. In this case we have two, so I thought on separating them.

> - Fixed a compile failure due to "Can't resolve 'crypto'" in Angular. [Issue #13267](https://github.com/Azure/azure-sdk-for-js/issues/13267).
> - Fixed an issue that the download stream returned by `BlobClient.download` won't release underlying resources unless it's fully consumed. [Issue #11850](https://github.com/Azure/azure-sdk-for-js/issues/11850).

Reviewer 8: Avoid giving out issue and PR links in the high level release notes. These are available in the changelog

```suggestion
- Fixed a compile time failure in resolving the crypto module in Angular projects which got introduced in a previous update. 
- Fixed an issue where the download stream returned by `BlobClient.download` method would not release underlying resources unless it's fully consumed.
```

##### New Features in @azure/storage-blob@12.5.0-beta.1 ([Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-blob/CHANGELOG.md#1250-beta1-2021-02-09))

- Now support Batch operations scoped to the Container level. You can use `ContainerClient.getBlobBatchClient()` to get such a `BlobBatchClient`.

#### @azure/storage-file-datalake [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-file-datalake/CHANGELOG.md)

##### Major Fixes in @azure/storage-file-datalake@12.3.1 ([Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-file-datalake/CHANGELOG.md#1231-2021-02-03))

> - Fixed a compile failure due to "Can't resolve 'crypto'" in Angular. [Issue #13267](https://github.com/Azure/azure-sdk-for-js/issues/13267).

Reviewer 8: Avoid giving out issue and PR links in the high level release notes. These are available in the changelog

```suggestion
- Fixed a compile time failure in resolving the crypto module in Angular projects which got introduced in a previous update.
```

### Azure Event Grid

We released a new beta package for the Event Grid service.

#### @azure/eventgrid [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventgrid/eventgrid/CHANGELOG.md#300-beta3-2020-10-06)

##### Breaking changes in @azure/eventgrid@3.0.0-beta.3

- The type definitions for SMS events sent by Azure Communication Services have been renamed, to use the prefix "AcsSms" instead of "Acssms". If you are using TypeScript and explicitly referencing these interfaces, you will need to update your code to use the new names. The payload of the events is unchanged.
- `EventGridSharedAccessCredential` has been removed, in favor of `AzureSASCredential`. Code which is using `EventGridSharedAccessCredential` should now use `AzureSASCredential` instead.
- When constructing the client, you must now include the schema type your topic is configured to expect (one of "EventGrid", "CloudEvent" or "Custom").
- The `sendEvents` methods have been collapsed into a single method on the client called `send` which uses the input schema that was configured on the client.

##### New Features in @azure/eventgrid@3.0.0-beta.3

> - Added distributed tracing support. EventGridProducerClient will now create spans when sending events to Event Grid.

Reviewer 8: ```suggestion
- Added distributed tracing support. `EventGridProducerClient` will now create spans when sending events to Event Grid.
```

> - Added distributed tracing support. EventGridProducerClient will now create spans when sending events to Event Grid.

Reviewer: ```suggestion
- Added distributed tracing support. EventGridProducerClient will now create spans when sending events to Event Grid.

### Azure Data Tables

We released a new beta package for the Data Tables service

#### @azure/data-tables [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/tables/data-tables/CHANGELOG.md#100-beta3-2021-02-09)

##### Breaking changes in @azure/data-tables@1.0.0-beta.5

- Migrated from  Core V1 (`@azure/core-http`) to Core v2 (`@azure/core-https` & `@azure/core-client`). With this change, the response types no longer contain the raw response `_response`. To access the raw response, an `onResponse` callback has to be passed in the request options bag.

   ```typescript
   let createResult: FullOperationResponse  | undefined;
   await client.createEntity(testEntity, { onResponse: (response) => (createResult = response) });
   ```
```

Author: Pushed here: https://github.com/Azure/azure-sdk/pull/2333/files#diff-a1ff6f8d1225133bd78843c3c728c052d26119ab995b81106191195e06d295a4R214

## Release highlights

---

=== COPY THIS AND ADD THE INFORMATION OF YOUR PACKAGE: ===

Keep in mind that:

- Including the package name in the headers makes the URL links work for multiple packages.
- The format of this file will be cleaned up once all of your proposals are in.

---

### _Project name_ 

#### @azure/_package-name_ [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/<service-folder>/<package-folder>/CHANGELOG.md)

(leave blank)

##### Breaking Changes on @azure/_package-name_@_version_

- _Add one or more, or remove the "Breaking Changes on ..." entire section._

##### New Features on @azure/_package-name_@_version_

- _Add one or more, or remove the "New Features on ..." section._

##### Major Fixes on @azure/_package-name_@_version_

- _Add one or more, or remove the "Major Fixes on ..." section._

---

> 

Reviewer 3: ```suggestion
### Event Hubs

#### @azure/event-hubs [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/CHANGELOG.md)

##### New Features on @azure/event-hubs@5.4.0

- You can now specify specify custom endpoint address as a client constructor option to use when communicating with the Event Hubs service, which is useful when your network does not allow communicating to the standard Event Hubs endpoint.

- Added a helper method `parseEventHubConnectionString` that can be used to validate and parse a given connection string for Azure Event Hubs.

##### Major Fixes on @azure/event-hubs@5.4.0

- Fixes an issue where the `RetryMode` enum for use when setting the RetryOptions.mode field in EventHubConsumerClientOptions or EventHubClientOptions wasn't exported by the package.

```

Author: Thank you! I have pushed an update here: https://github.com/Azure/azure-sdk/pull/2333/commits/ce3ac4b7e18b5327628227ac86cd9b97405f2b55

### Azure Identity

#### @azure/identity [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/identity/identity/CHANGELOG.md)

Identity is releasing a patch with a dependency fix and a bug fix, and a beta that includes MSAL 2.0 with PKCE support.

##### Major Fixes on @azure/identity@1.2.3

- Fixed Azure Stack support for the NodeJS version of the `InteractiveBrowserCredential`.
- The 'keytar' dependency has been updated to the latest version.

##### Breaking Changes on @azure/identity@1.2.4-beta.1

- In this beta we've updated `InteractiveBrowserCredential` to use the Auth Code Flow with PKCE rather than Implicit Grant Flow by default in the browser, to better support browsers with enhanced security restrictions.

> 

Reviewer 4: ```suggestion

### Azure Form Recognizer

#### @azure/ai-form-recognizer [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/formrecognizer/ai-form-recognizer/CHANGELOG.md)

##### Breaking Changes on @azure/ai-form-recognizer@3.1.0-beta.2

- Reworked the type of the `appearance` field and its children of the `FormLine` type to have more specific names.

```

@sadasant This is only a breaking change with respect to the previous beta `3.1.0-beta.1`. Should it still be filed as a breaking change here?

Author: @witemple-msft up to you! How would it be better for our customers? I can think of arguments on favor and against using breaking change here. It feels like since this is a beta it would be better to state it as an update... but I'm not strongly on favor of it!

Author: On the other hand, these titles are a bit arbitrary! So I don't mind changing one of them. I'll set that header as "Changes" instead of "Breaking changes". Let me know if that sounds ok or not 🌞

Author: I have pushed an update here: https://github.com/Azure/azure-sdk/pull/2333/commits/ce3ac4b7e18b5327628227ac86cd9b97405f2b55

> 

Reviewer 5: ```suggestion

### Azure Storage

We released hot fix versions and also beta versions for new service features in Azure Storage Service API version 2020-06-10.

#### @azure/storage-blob [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-blob/CHANGELOG.md)

##### New Features on @azure/storage-blob@12.5.0-beta.1

- Now support Batch operations scoped to the Container level. You can use `ContainerClient.getBlobBatchClient()` to get such a `BlobBatchClient`.

##### Major Fixes on @azure/storage-blob@12.4.1

- Fixed a compile failure due to "Can't resolve 'crypto'" in Angular. [Issue #13267](https://github.com/Azure/azure-sdk-for-js/issues/13267).
- Fixed an issue that the download stream returned by `BlobClient.download` won't release underlying resources unless it's fully consumed. [Isssue #11850](https://github.com/Azure/azure-sdk-for-js/issues/11850).

#### @azure/storage-file-datalake [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-file-datalake/CHANGELOG.md)

##### New Features on @azure/storage-file-datalake@12.4.0-beta.1

- Added support for Container Soft Delete. You can restore a deleted filesystem via `DataLakeServiceClient.undeleteFileSystem()`. And the `DataLakeServiceClient.listFileSystems()` now support an `includeDeleted` option to include soft deleted filesystems in the response.

##### Major Fixes on @azure/storage-file-datalake@12.3.1

- Fixed a compile failure due to "Can't resolve 'crypto'" in Angular. [Issue #13267](https://github.com/Azure/azure-sdk-for-js/issues/13267).

#### storage-file-share [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-file-share/CHANGELOG.md)

##### @azure/storage-file-share@12.4.1

This release contains bug fixes to improve quality.

#### storage-queue [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-queue/CHANGELOG.md)

##### @azure/storage-queue@12.3.1

This release contains bug fixes to improve quality.

```

Author: Thank you! I have pushed an update here: https://github.com/Azure/azure-sdk/pull/2333/commits/ce3ac4b7e18b5327628227ac86cd9b97405f2b55

#### @azure/storage-file-datalake [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-file-datalake/CHANGELOG.md)

##### Major Fixes in @azure/storage-file-datalake@12.3.1

- Fixed a compile failure due to "Can't resolve 'crypto'" in Angular. [Issue #13267](https://github.com/Azure/azure-sdk-for-js/issues/13267).

##### New Features in @azure/storage-file-datalake@12.4.0-beta.1

- Added support for Container Soft Delete. You can restore a deleted filesystem via `DataLakeServiceClient.undeleteFileSystem()`. And the `DataLakeServiceClient.listFileSystems()` now support an `includeDeleted` option to include soft deleted filesystems in the response.

#### storage-file-share [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-file-share/CHANGELOG.md)

##### @azure/storage-file-share@12.4.1

This release contains bug fixes to improve quality.

#### storage-queue [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-queue/CHANGELOG.md)

##### @azure/storage-queue@12.3.1

> This release contains bug fixes to improve quality.

Reviewer 9: ```suggestion
This release contains bug fixes to improve quality.

### Azure Event Grid

We released a new beta package for the Event Grid service.

#### @azure/eventgrid [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventgrid/eventgrid/CHANGELOG.md)

##### Breaking changes in @azure/eventgrid@3.0.0-beta.3

- The type definitions for SMS events sent by Azure Communication Services have been renamed, to use the prefix "AcsSms" instead of "Acssms". If you are using TypeScript and explicitly referencing these interfaces, you will need to update your code to use the new names. The payload of the events is unchanged.
- `EventGridSharedAccessCredential` has been removed, in favor of `AzureSASCredential`. Code which is using `EventGridSharedAccessCredential` should now use `AzureSASCredential` instead.
- When constructing the client, you must now include the schema type your topic is configured to expect (one of "EventGrid", "CloudEvent" or "Custom").
- The `sendEvents` methods have been collapsed into a single method on the client called `send` which uses the input schema that was configured on the client.

##### New Features in @azure/eventgrid@3.0.0-beta.3

- Added distributed tracing support. EventGridProducerClient will now create spans when sending events to Event Grid.
```

Author: Thanks! I'll add it in a bit 🌞

Author: Added it! Please review again 🙏

- Added a helper method `parseEventHubConnectionString` that can be used to validate and parse a given connection string for Azure Event Hubs.

##### Major Fixes in @azure/event-hubs@5.4.0

- Fixes an issue where the `RetryMode` enum for use when setting the RetryOptions.mode field in `EventHubConsumerClientOptions` or `EventHubClientOptions` wasn't exported by the package.

### Azure Identity

#### @azure/identity [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/identity/identity/CHANGELOG.md)

Identity is releasing a patch with a dependency fix and a bug fix, and a beta that includes MSAL 2.0 with PKCE support.

##### Major Fixes in @azure/identity@1.2.3

- Fixed Azure Stack support for the NodeJS version of the `InteractiveBrowserCredential`.
- The 'keytar' dependency has been updated to the latest version.

##### Changes in @azure/identity@1.2.4-beta.1

- In this beta we've updated `InteractiveBrowserCredential` to use the Auth Code Flow with PKCE rather than Implicit Grant Flow by default in the browser, to better support browsers with enhanced security restrictions.

> 

Reviewer 10: ```suggestion

### Azure Key Vault Admin [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/keyvault/keyvault-admin/CHANGELOG.md)

- We've added support for RBAC in Managed HSM instances. Users can now use the administration client to create custom role definitions and create role assignments.

### Azure Key Vault Keys [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/keyvault/keyvault-keys/CHANGELOG.md)

- Added a constructor overload to CryptographyClient that takes a JsonWebKey and allows for local-only subset of operations.
- Added a createOctKey convenience method to create a key of type oct or oct-HSM as appropriate.

```

Something like this? It's my first time! 😄

Author: More like this https://github.com/Azure/azure-sdk/pull/2333/commits/ba6893286c3b904ce4870db1c3012e559a835ce7#diff-a1ff6f8d1225133bd78843c3c728c052d26119ab995b81106191195e06d295a4R78-R94

Are we not releasing certificates and secrets? 🤔

Reviewer 10: Done!

Author: nice!!! thanks!

#### @azure/storage-file-datalake [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-file-datalake/CHANGELOG.md)

##### Major Fixes in @azure/storage-file-datalake@12.3.1

- Fixed a compile failure due to "Can't resolve 'crypto'" in Angular. [Issue #13267](https://github.com/Azure/azure-sdk-for-js/issues/13267).

##### New Features in @azure/storage-file-datalake@12.4.0-beta.1

- Added support for Container Soft Delete. You can restore a deleted filesystem via `DataLakeServiceClient.undeleteFileSystem()`. And the `DataLakeServiceClient.listFileSystems()` now support an `includeDeleted` option to include soft deleted filesystems in the response.

#### storage-file-share [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-file-share/CHANGELOG.md)

##### @azure/storage-file-share@12.4.1

This release contains bug fixes to improve quality.

#### storage-queue [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-queue/CHANGELOG.md)

##### @azure/storage-queue@12.3.1

This release contains bug fixes to improve quality.

> ## Latest Releases

Reviewer 11: ```suggestion
### Azure Text Analytics

#### @azure/ai-text-analytics [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/textanalytics/ai-text-analytics/CHANGELOG.md)

##### New Features in @azure/ai-text-analytics@5.1.0-beta.4

- A new option to control how the offset is calculated by the service, `stringIndexType`, is added to `analyzeSentiment`, `recognizeEntities`, `recognizePiiEntities`, and `beginAnalyzeHealthcareEntities`. Furthermore, `stringIndexType` is added to task types `RecognizeEntitiesAction` and `RecognizePiiEntitiesAction`, which are the types of input actions to the `beginAnalyzeBatchActions` method. For more information, see [the Text Analytics documentation](https://docs.microsoft.com/azure/cognitive-services/text-analytics/concepts/text-offsets#offsets-in-api-version-31-preview).
- The poller for the `beginAnalyzeBatchActions` long-running operation gained the ability to return certain metadata information about the currently running operation (e.g., when the operation was created, will be expired, and last time it was updated, and also how many actions completed and failed so far). Also, the poller for `beginAnalyzeHealthcareEntities` gained a similar ability.

##### Breaking Changes on @azure/ai-text-analytics@5.1.0-beta.4

- `beginAnalyzeHealthcare` is renamed to `beginAnalyzeHealthcareEntities`.
- `beginAnalyze` is renamed to `beginAnalyzeBatchActions`.
- The healthcare entities returned by `beginAnalyzeHealthcare` are now organized as a directed graph where the edges represent a certain type of healthcare relationship between the source and target entities. Edges are stored in the `relatedEntities` property.
- The `links` property of `HealthcareEntity` is renamed to `dataSources`, a list of objects representing medical databases, where each object has `name` and `entityId` properties.
- the words "operation" and "action" are used consistently in our names and documentation instead of "job" and "task" respectively.

## Latest Releases
```

Author: Can we make this shorter? It should not be exactly like the changelog entry. 🙏

Author: Woah! thanks!

Author: I have added it! Thank you!
